src/map/map.c

```diff
@@ -2945,6 +2945,45 @@ static enum unit_dir map_calc_dir(const struct block_list *src, int16 x, int16 y
 	return dir;
 }
 
+/**
+ * Randomizes (x, y) to a walkable cell on a rectangle with (x, y) being the center cell.
+ *
+ * __Only__ tries each quarter of the rectangle once, doesn't try all possibilities. (Aegis behavior)
+ * The rectangle has a width of 2 * x_range - 1 and a height of 2 * y_range - 1 .
+ *
+ * @remark e.g. (x + x_range, y + y_range) is not part of the rectangle and won't be randomized.
+ * @param[in] m map ID for the walkable cell checks
+ * @param[in,out] x pointer which contains the x-axis center value of the rectangle
+ * @param[in,out] y pointer which contains the y-axis center value of the rectangle
+ * @param[in] x_range horizontal distance from center to border
+ * @param[in] y_range vertical distance from center to border
+ * @retval 0 success, random walkable cell found
+ * @retval 1 failure, no cell found, x and y remain unchanged
+ */
+static int map_get_random_cell_in_range(int16 m, int16 *x, int16 *y, int16 x_range, int16 y_range)
```

Comment:
for advanced usage of walking code, here need also bl parameter of source who want to move.
this allow in plugin extend walking code in many ways
this bl should be used in ``map->getcell`` and  ``path->search``

Reply:
The functions are intentionally made in a way that you're not fixed to a bl. If for some reason you simply need a random cell, without bothering with a bl, you should get one. Definitely won't add a non-nullable bl param in both new functions due to that reason.
Plugins could just hook a different function imo, I'd rather not bloat these 2 new functions with an unnecessary parameter just because there are some edge cases of plugins that might be able to use it.

Reply:
well all other walking related functions support bl as parameter.

``map_get_random_cell_in_range`` function search random cell to spawn or move some one here (based on bl).
and code may find different free cells based on bl already. it works really nice.

for example mobs/npcs/players may have attributes to walk on water, or walk (fly) on some walls. this features not in hercules because walk maps based on gravity client walk maps, and here they limited. but with custom plugin walk maps can be custom too...

for your changes need simply introduce ``bl`` parameter here, and set it to most correct value. if source object unknown, then set it to ``NULL``, but i think most code related to move/spawn some ``bl``.

Reply:
Hmm, that's a convincing point. I will change it.